tests/unit/plugins/mesos/timeout_executor_test.py

```diff
@@ -13,15 +17,136 @@ def mock_Thread():
 
 
 @pytest.fixture
-def mock_downstream():
-    return mock.MagicMock()
+def source_queue():
+    return Queue()
+
+
+@pytest.fixture
+def mock_downstream(source_queue):
+    executor = mock.MagicMock()
+    executor.get_event_queue.return_value = source_queue
+    return executor
 
 
 @pytest.fixture
 def mock_timeout_executor(mock_Thread, mock_downstream):
     return TimeoutExecutor(downstream_executor=mock_downstream)
 
 
+@pytest.fixture
+def mock_task_config():
+    return MesosTaskConfig(
+        uuid='mock_uuid',
+        name='mock_name',
+        image='mock_image',
+        cmd='mock_cmd',
+        timeout=1000,
+    )
+
+
+@pytest.fixture
+def mock_entry(mock_task_config):
+    return TaskEntry(
+        task_id=mock_task_config.task_id,
+        deadline=mock_task_config.timeout + 2000,
+    )
+
+
+@pytest.fixture
+def mock_event(mock_task_config):
+    return Event(
+        kind='task',
+        timestamp=1234.5678,
+        terminal=True,
+        task_id=mock_task_config.task_id,
+        platform_type='mesos',
+        message='mock_message',
+        task_config=mock_task_config,
+        raw='raw_event',
+    )
+
+
+# timeout_loop #################################################################
+def test_timeout_loop_nontask(
+    mock_timeout_executor,
+    mock_event,
+):
+    mock_event = mock_event.set('kind', 'control')
+    mock_entry = TaskEntry('different_id', deadline=1234)
+    mock_timeout_executor.stopping = True
+    mock_timeout_executor.src_queue = Queue()
+    mock_timeout_executor.src_queue.put(mock_event)
+    mock_timeout_executor.running_tasks.append(mock_entry)
+    mock_timeout_executor.downstream_executor.kill = mock.Mock()
+    time.time = mock.Mock(return_value=0)
+
+    mock_timeout_executor.timeout_loop()
+
+    assert mock_timeout_executor.downstream_executor.kill.call_count == 0
```

Comment:
Again, we should make this simpler and just assert the control event is on the queue.

Reply:
Sure.,